core/src/connection.rs

```diff
@@ -19,6 +19,9 @@
 // DEALINGS IN THE SOFTWARE.
 
 use crate::multiaddr::{Multiaddr, Protocol};
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+static NEXT_CONNECTION_ID: AtomicUsize = AtomicUsize::new(0);
 
 /// Connection identifier.
 #[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
```suggestion
#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
```

I suggest the only guarantee that `ConnectionId` provides is uniqueness, not monotonic increasing. Thus there is no need for ordering. Is any existing functionality depending on this? If not, how about removing it right away?

Reply:
Yeah happy to remove it.